prov/rxm/src/rxm_ep.c

```diff
@@ -284,106 +284,114 @@ static void rxm_recv_queue_close(struct rxm_recv_queue *recv_queue)
 	// TODO cleanup recv_list and unexp msg list
 }
 
-static int rxm_ep_txrx_res_open(struct rxm_ep *rxm_ep)
+static int rxm_ep_txrx_pool_create(struct rxm_ep *rxm_ep)
 {
+	size_t i;
 	int ret;
 
-	FI_DBG(&rxm_prov, FI_LOG_EP_CTRL,
-	       "MSG provider mr_mode & FI_MR_LOCAL: %d\n",
-	       rxm_ep->msg_mr_local);
-
-	ret = rxm_buf_pool_create(rxm_ep,
-				  rxm_ep->msg_info->tx_attr->size,
-				  rxm_ep->rxm_info->tx_attr->inject_size +
-				  sizeof(struct rxm_tx_buf),
-				  &rxm_ep->buf_pools[RXM_BUF_POOL_TX_MSG],
-				  RXM_BUF_POOL_TX_MSG);
-	if (ret)
-	        return ret;
-
-	ret = rxm_buf_pool_create(rxm_ep,
-				  rxm_ep->msg_info->tx_attr->size,
-				  rxm_ep->rxm_info->tx_attr->inject_size +
-				  sizeof(struct rxm_tx_buf),
-				  &rxm_ep->buf_pools[RXM_BUF_POOL_TX_TAGGED],
-				  RXM_BUF_POOL_TX_TAGGED);
-	if (ret)
-	        goto err1;
-
-	ret = rxm_buf_pool_create(rxm_ep,
-				  rxm_ep->msg_info->tx_attr->size,
-				  rxm_ep->rxm_info->tx_attr->inject_size +
-				  sizeof(struct rxm_tx_buf),
-				  &rxm_ep->buf_pools[RXM_BUF_POOL_TX_ACK],
-				  RXM_BUF_POOL_TX_ACK);
-	if (ret)
-	        goto err2;
-
-	ret = rxm_buf_pool_create(rxm_ep,
-				  rxm_ep->msg_info->tx_attr->size,
-				  rxm_ep->rxm_info->tx_attr->inject_size +
-				  sizeof(struct rxm_tx_buf),
-				  &rxm_ep->buf_pools[RXM_BUF_POOL_TX_LMT],
-				  RXM_BUF_POOL_TX_LMT);
-	if (ret)
-	        goto err3;
-
 	ret = rxm_buf_pool_create(rxm_ep,
 				  rxm_ep->msg_info->rx_attr->size,
 				  rxm_ep->rxm_info->tx_attr->inject_size +
 				  sizeof(struct rxm_rx_buf),
 				  &rxm_ep->buf_pools[RXM_BUF_POOL_RX],
 				  RXM_BUF_POOL_RX);
 	if (ret)
-		goto err4;
+		return ret;
 	dlist_init(&rxm_ep->post_rx_list);
 	dlist_init(&rxm_ep->repost_ready_list);
 
+	/* Allocates resources for TX pools */
+	for (i = RXM_BUF_POOL_TX_MSG; i <= RXM_BUF_POOL_TX_LMT; i++) {
```

Comment:
I would actually init i = 0, rather than assuming TX_MSG is the first index.

Reply:
`RXM_BUF_POOL_TX_MSG` has index = 1
`RXM_BUF_POOL_RX` has index = 0.

Do you mean to change them, and move RX to the end of enum?

Reply:
I didn't realize that.  In that case, maybe add a new enum value like:
```
RXM_BUF_POOL_TX_START,
RXM_BUF_POOL_TX_MSG = RXM_BUF_POOL_TX_START,
...
```

Reply:
Ok, sounds good!
Done